### PR TITLE
Issue #9950: Ordered Aggregate Performance

### PR DESCRIPTION
**Pull request by hawkfish**
Reduce memory requirements and increase performance.

* Add linked list merging
* Add Swap to reduce vector overhead
* Fix PhysicalUngroupedAggregate to manage ArenaAllocator lifetimes.

fixes: #9980
fixes: duckdblabs/duckdb-internal#892

### CODE REVIEW

**Review by lnkuiper — APPROVED** [review]
LGTM. Great that you were able to solve the problem!

I'd like to look into more optimizations in the future (as discussed in the internal issue), when I have the time :)

**Review by Mytherin — COMMENTED** [review]
Thanks! LGTM - one minor comment:

File: benchmark/micro/aggregate/ordered_first.benchmark [hunk 1]
```diff
@@ -16,4 +16,4 @@ SELECT SUM(agg) FROM (
 )
 
 result I
-24420932461
+49999995000000
```
[the remark this answers is not in the document]
Comment (hawkfish) [reply benchmark/micro/aggregate/ordered_first.benchmark:19]: Wow - that looks wrong. And I have no memory of changing it!

[the remark this answers is not in the document]
Comment (hawkfish) [reply benchmark/micro/aggregate/ordered_first.benchmark:19]: Oh the test is just wrong.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**